### PR TITLE
Champion: display a bottom sheet

### PR DESCRIPTION
**Pull request by leandroalonso**
## To test

1. Run the app on a real device
2. Go to Profile > Settings > Developer > Set to Lifetime
3. Go to Profile > tap Account
4. Tap "Pocket Casts Champion"
5. Tap "Rate Pocket Casts"
6. ✅ This should open the review screen
7. Go back to the app
8. Go to Profile > Settings > Developer > Set to Paid Plus
9. Go to Profile > tap Account
10. Tap "Monthly"
11. ✅ Nothing should happen

### Themes

You can either test themes one by one or by opening `ChampionView`, enabling preview and checking one by one.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift [hunk 1]
```diff
@@ -51,32 +64,38 @@ struct AccountHeaderView: View {
...
+                expirationLabel(),
+                nil
             )
         case .lifetime:
             // Lifetime membership, show a thank you message
             return (
                 L10n.subscriptionsThankYou,
                 Text(L10n.plusChampion)
-                    .foregroundColor(theme.green)
+                    .foregroundColor(theme.green), {
```
Comment (danielebogo) [inline podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift:89]: Nit: Better to start using the `foregroundStyle` as this is deprecated

File: podcasts/Profile - SwiftUI/Account/ChampionView.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @EnvironmentObject var theme: Theme
+
+    @Environment(\.openURL) private var openURL
+
+    var body: some View {
+        VStack(spacing: 16) {
+            Image("crown")
+                .renderingMode(.template)
+                .tint(theme.primaryUi05Selected)
+            Text("You’re a true champion of Pocket Casts!")
```
Comment (danielebogo) [inline podcasts/Profile - SwiftUI/Account/ChampionView.swift]: Should we localize this string? 🤔

File: podcasts/Profile - SwiftUI/Account/ChampionView.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                .font(style: .title, weight: .semibold)
+                .multilineTextAlignment(.center)
+            Text("Thanks for being with us since the beginning! If you enjoy using our app, we’d love to hear your feedback.")
```
Comment (danielebogo) [inline podcasts/Profile - SwiftUI/Account/ChampionView.swift]: Same here about localization

File: podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift [hunk 4]
```diff
@@ -28,21 +30,32 @@ struct AccountHeaderView: View {
...
                     .font(size: 14, style: .subheadline, weight: .medium)
+                    .sheet(isPresented: $showingChampion) {
+                        ZStack {
+                            theme.primaryUi01.ignoresSafeArea()
+
+                            if #available(iOS 16.0, *) {
+                                ChampionView()
+                                    .presentationDetents([.medium])
+                            } else {
+                                ChampionView()
```
Comment (leandroalonso) [inline podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift:49]: On iOS 15 there won't be a bottom sheet and instead a big screen. Far from ideal but I think this is an ok compromise as ~2% of users are on it and this is not a core flow/screen.

Comment (danielebogo) [reply podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift:49]: It makes sense as making a host vc just to set the detents doesn't make sense

File: podcasts/Subscription.xcassets/crown.imageset/crown.svg
Comment (danielebogo) [inline podcasts/Subscription.xcassets/crown.imageset/crown.svg:1]: I think the image should not have the empty sapce on the right, but being cropped on the crown edge. Same on the top.

**Review by danielebogo — APPROVED** [review]
Hey @leandroalonso I followed your steps and it worked as expected. I left a nit only about the usage of `foregroundStyle`.

However I have another nit about the image alignment. I think David aligned it based on the PC red circle than the whole image (which includes the crown).

File: podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift (hunk 1 above)
Comment (leandroalonso) [reply podcasts/Profile - SwiftUI/Account/AccountHeaderView.swift:89]: Hmm, I guess it depends on the `View`. For `Text` is not and `foregroundStyle` is available only for iOS 17:

Screenshot 2024-07-29 at 15 07 55

File: podcasts/Subscription.xcassets/crown.imageset/crown.svg
Comment (leandroalonso) [reply podcasts/Subscription.xcassets/crown.imageset/crown.svg:1]: Good point. The original `SVG` had the padding at the right though. I'll check if this is intended or not.

Comment (leandroalonso) [reply podcasts/Subscription.xcassets/crown.imageset/crown.svg:1]: Updated! Plus the crown was gold while it should be red.

**Pull request merged**